### ISSUE TITLE
Scatter plots with log scaling

### ISSUE THREAD

**Issue by Phlya** [matplotlib/matplotlib#15978]
When using `plt.scatter` with log-scaled axes, order of calls is important: if axes are set to log scale before `plt.scatter` is called, margins of the plot are huge:

```python
import matplotlib.pyplot as plt
import numpy as np
f, ax = plt.subplots()
ax.set_xscale('log')
ax.set_yscale('log')
ax.scatter(2**np.arange(10), 2**np.arange(10))
```

**Actual outcome**
![image](https://user-images.githubusercontent.com/2895034/71193826-af474d80-2282-11ea-8710-8074b557e209.png)

**Expected outcome**
When `plt.scatter` is called before axes are set to log scale, it looks normal:
```python
import matplotlib.pyplot as plt
import numpy as np
f, ax = plt.subplots()
ax.scatter(2**np.arange(10), 2**np.arange(10))
ax.set_xscale('log')
ax.set_yscale('log')
```
![image](https://user-images.githubusercontent.com/2895034/71193927-e0278280-2282-11ea-8cc2-2aee32e358f6.png)

  * Operating system: Ubuntu 18.04
  * Matplotlib version: 3.1.1
  * Matplotlib backend (`print(matplotlib.get_backend())`): module://ipykernel.pylab.backend_inline
  * Python version: 3.8
  * Jupyter version (if applicable): Jupyter Lab 1.1.4

In some earlier version of matplotlib (or something else) it worked as expected with any order of calls, but now I am getting this problem.

**Comment by anntzer** [comment]
This is fixed as of master, AFAICT.

**Comment by QuLogic** [comment]
This was fixed by #13642 and in 3.2.0.
